dattri/task.py

```diff
@@ -142,17 +152,23 @@ def get_grad_target_func(
         if layer_name is not None:
             error_msg = "layer_name has not been implemented yet."
             raise NotImplementedError(error_msg)
+
+        if self.grad_func_kwargs != {"in_dims": in_dims}:
+            self.grad_func = vmap(grad(self.target_func), in_dims=in_dims)
```

Comment:
do we test these different `in_dims`?

Reply:
No, but it's directly called by the attributors, so I think it's fine for now. If the `in_dims` is not correct, an error will be reported by `vmap`.